2.4/Dockerfile

```
@@ -220,6 +220,7 @@ RUN set -eux; \
 	find /usr/local -type f -executable -exec ldd '{}' ';' \
 		| awk '/=>/ { print $(NF-1) }' \
 		| sort -u \
+		| xargs -r basename -a \
```

Similar comment to https://github.com/docker-library/python/pull/822/files#r1227154811:

```suggestion
		| awk '/=>/ { so = $(NF-1); if (index(so, "/usr/local/") == 1) { next }; gsub("^/(usr/)?", "", so); print so }' \
		| sort -u \
```

(I'm happy to take over from here if you'd prefer :+1:)

---

Comment is not available anymore.

I'll try to understand why you wan't to skip shared objects inside `/usr/local`.

Otherwise rely on base name instead of an absolute path transformed in relative path may be more robust in time. What do you think ?

---

Doh, sorry - GitHub did us dirty.  I've updated the link above to https://github.com/docker-library/python/pull/822#discussion_r1227154811, which might answer your question.

The reason I'd prefer not to rely on `basename` here and instead be as specific as we can reasonably be while still being generic enough to match both `/usr/something` and `/something` is that something like `foo.so` might match inside other things like man pages, other directories, etc, and the purpose of this code is to _only_ match things installed by packages, and to do so reliably so we can keep the proper library dependencies installed (and _only_ the necessary dependencies), which is also why we should skip any matches in `/usr/local` (which I don't _think_ we have any of right now, and they'll generally just return nothing in `dpkg-query --search`, but we might as well exclude them explicitly just to be sure and to put less work on `dpkg-query --search`).

---

Ok I update commit in this way